templates/15_insert.go.tpl

```diff
@@ -201,3 +201,173 @@ CacheNoHooks:
 	return nil
 	{{- end}}
 }
+
+// InsertAll inserts all rows with the specified column values, using an executor.
+func (o {{$alias.UpSingular}}Slice) InsertAll({{if .NoContext}}exec boil.Executor{{else}}ctx context.Context, exec boil.ContextExecutor{{end}}, columns boil.Columns) error {
+	ln := int64(len(o))
+	if ln == 0 {
+		return nil
+	}
+ 	var sql, retQuery string
+	vals := []interface{}{}
+	retMaps := make([][]interface{}, 0)
+	for i, row := range o {
+		if err := row.doBeforeInsertHooks(ctx, exec); err != nil {
+			return err
+		}
+ 		{{- if not .NoAutoTimestamps -}}
+		{{- $colNames := .Table.Columns | columnNames -}}
+		{{if containsAny $colNames "created_at" "updated_at"}}
+		currTime := time.Now().In(boil.GetLocation())
+		{{range $ind, $col := .Table.Columns}}
+		{{- if eq $col.Name "created_at" -}}
+		{{- if eq $col.Type "time.Time" }}
+		if row.CreatedAt.IsZero() {
+			row.CreatedAt = currTime
+		}
+		{{- else}}
+		if queries.MustTime(row.CreatedAt).IsZero() {
+			queries.SetScanner(&row.CreatedAt, currTime)
+		}
+		{{- end -}}
+		{{- end -}}
+		{{- if eq $col.Name "updated_at" -}}
+		{{- if eq $col.Type "time.Time"}}
+		if row.UpdatedAt.IsZero() {
+			row.UpdatedAt = currTime
+		}
+		{{- else}}
+		if queries.MustTime(row.UpdatedAt).IsZero() {
+			queries.SetScanner(&row.UpdatedAt, currTime)
+		}
+		{{- end -}}
+		{{- end -}}
+		{{end}}
+		{{end}}
+		{{- end}}
+ 		nzDefaults := queries.NonZeroDefaultSet({{$alias.DownSingular}}ColumnsWithDefault, row)
+ 		wl, returnColumns := columns.InsertColumnSet(
+			{{$alias.DownSingular}}Columns,
+			{{$alias.DownSingular}}ColumnsWithDefault,
+			{{$alias.DownSingular}}ColumnsWithoutDefault,
+			nzDefaults,
+		)
+ 		if i == 0 {
+			sql = "INSERT INTO {{$schemaTable}} " + "({{.LQ}}" + strings.Join(wl, "{{.RQ}},{{.LQ}}") + "{{.RQ}})" + " VALUES "
+			{{if not .Dialect.UseLastInsertID -}}
+			retQuery = " RETURNING {{.LQ}}" + strings.Join(returnColumns, "{{.RQ}},{{.LQ}}") + "{{.RQ}} "
+			{{else -}}
+			retQuery = fmt.Sprintf("SELECT {{.LQ}}%s{{.RQ}} FROM {{$schemaTable}} WHERE {{.LQ}}%s{{.RQ}} IN %s",
```

Comment:
`select ... in ...` cannot make the same order between your primary key and return record.

Therefore, i think you have to add a `ORDER BY` PK clause to keep the order.